recipes/packages.rb

```diff
@@ -15,9 +15,26 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+include_recipe 'yum-ius' if node['osl-php']['use_ius']
 
-if node['osl-php']['packages'].any?
-  node.normal['php']['packages'] = node['osl-php']['packages']
+# Get package prefix from version (e.g. "php71u" or "php")
+prefix = if node['osl-php']['use_ius']
+           'php' + node['php']['version'].split('.')[0, 2].join + 'u'
+         else
+           'php'
+         end
+
+php_packages = node['osl-php']['packages']
+
+# Prepend PHP package prefix to short packages (e.g. "php71u-memcached")
+if node['osl-php']['short_packages'].any?
+  php_packages += node['osl-php']['short_packages']
+                  .map { |pkg| prefix + '-' + pkg }
+end
+
+if php_packages.any? || node['osl-php']['use_ius']
+  node.default['php']['packages'] = php_packages
+  node.default['php']['packages'] <<= prefix # Also the name of the main package
```

Comment:
I noticed that with later versions `php71u` is not a real package, `mod_php71u` [just provides it.](https://ius.io/FAQs/#i-dont-see-php70u-in-the-repos-where-is-it)

I don't know if there's any harm in treating the prefix as the main PHP package, for that reason, but if you think we should do it differently let me know.